optuna/visualization/_pareto_front.py

```diff
@@ -386,6 +384,11 @@ def _make_scatter_object(
             marker=marker,
             showlegend=False,
         )
+    else:
+        raise ValueError(
+            "`plot_pareto_front` function only supports 2 or 3 targets."
+            " you used {} targets now.".format(n_targets)
+        )
```

Comment:
It looks that the number of targets are validated in `_get_pareto_front_info`. Thus these are unreachable lines. How about replacing these lines like the following?

```suggestion
    assert False, "Must not reach here"
```

Reply:
Alternatively, I'm wondering if we could move `ValueError` part into `_get_pareto_front_info` not to repeat the same error message in different files.

Reply:
Oh, I didn't notice that point :bow:

> Alternatively, I'm wondering if we could move ValueError part into _get_pareto_front_info not to repeat the same error message in different files.

Yes, I think that makes sense 👍

Reply:
Thanks! I've moved the value error part as in https://github.com/optuna/optuna/pull/3546/commits/a0a693351f7ba1f5c543d0fa9a922da30371b158.